src/modules/map/components/cesium/CesiumKMLLayer.vue

```diff
@@ -38,26 +36,21 @@ export default {
     computed: {
         ...mapState({
             availableIconSets: (state) => state.drawing.iconSets,
+            projection: (state) => state.position.projection,
         }),
     },
     methods: {
         async loadLayer() {
             try {
                 const kml = await getKmlFromUrl(this.url)
-                const features = new KML().readFeatures(kml, {
-                    // Reproject all features to webmercator, as this is the projection used for the view
-                    featureProjection: WEBMERCATOR.epsg,
-                })
+                const features = parseKml(kml, this.projection, this.availableIconSets)
                 if (features) {
                     this.olLayer.setSource(new VectorSource({ wrapX: true }))
-                    features.forEach((olFeature) => {
-                        EditableFeature.deserialize(olFeature, this.availableIconSets)
-                    })
                     // remove all old features first
                     this.olLayer.getSource().clear()
                     // add the deserialized features
                     this.olLayer.getSource().addFeatures(features)
-                    return WEBMERCATOR.epsg
```

Comment:
is it really necessary that we return a String containing the EPSG? That sounds a bit weird

Reply:
Good question I did not check

Reply:
It is indeed used and needed.